### PR TITLE
refactor(editor): Migrate `tags.store` to use composition API (no-changelog)

### PR DESCRIPTION
**Pull request by RicardoE105**
## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/editor-ui/src/stores/tags.store.ts [hunk 1]
```diff
@@ -1,105 +1,119 @@
...
-			this.tags = rest;
-		},
-
-		async fetchAll(params?: { force?: boolean; withUsageCount?: boolean }): Promise<ITag[]> {
-			const { force = false, withUsageCount = false } = params || {};
-			if (!force && this.fetchedAll && this.fetchedUsageCount === withUsageCount) {
-				return Object.values(this.tags);
-			}
+export const useTagsStore = defineStore(STORES.TAGS, () => {
+	const tags = ref<{ [id: string]: ITag }>({});
```
Comment (alexgrozav) [inline packages/editor-ui/src/stores/tags.store.ts:10]: nit: Typically `Record<string, ITag>` is the preferred way to type key value pairs.

File: packages/editor-ui/src/stores/tags.store.ts [hunk 2]
```diff
@@ -1,105 +1,119 @@
...
+		return Object.values(tags.value).sort((a, b) => a.name.localeCompare(b.name));
+	});
+
+	const isLoading = computed(() => loading.value);
+
+	const hasTags = computed(() => Object.keys(tags.value).length > 0);
+
+	const getTagById = computed(() => {
+		return (id: string) => tags.value[id];
+	});
```
Comment (alexgrozav) [inline packages/editor-ui/src/stores/tags.store.ts:30]: Computed getters will break in composition API if used within another computed value. We should just rename `tags` to `tagsById`, and reference the value directly instead (i.e. `tagsStore.tagsById[id]`).

File: packages/editor-ui/src/stores/tags.store.ts [hunk 3]
```diff
@@ -1,105 +1,119 @@
...
+
+	// Methods
+
+	const setAllTags = (loadedTags: ITag[]) => {
+		tags.value = loadedTags.reduce((accu: { [id: string]: ITag }, tag: ITag) => {
```
Comment (alexgrozav) [inline packages/editor-ui/src/stores/tags.store.ts:35]: nit: Would prefer `Record<string, ITag>` instead

**Review by alexgrozav — CHANGES_REQUESTED** [review]
Looking good! Small issue with the computed getter.

**Review by alexgrozav — APPROVED** [review]
Awesome! Thank you for making the changes!

**Review by alexgrozav — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.49.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.49.0)

**Pull request merged**